kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java

```diff
@@ -79,9 +76,12 @@
         .withQos( mqttConsumerMeta.getQos() )
         .withStep( mqttConsumer )
         .withCallback( callback )
+        .withUsername( mqttConsumerMeta.getUsername() )
```

Comment:
Should these be environment substituted?

Reply:
yes!